utils/get_latest_copr_build

```diff
@@ -2,83 +2,89 @@
+        error += '\n'
+    sys.stderr.write(error)
+    # dump ENV dictionary
+    sys.stderr.write('Passed (or default) environment variables:\n')
+    for var, value in ENV_VARS.items():
+        sys.stderr.write('  {}: {}\n'.format(var, value))
+    sys.exit(1)
 
-if '--debug' in sys.argv:
-    json.dump(builds, sys.stderr, sort_keys=True, indent=2)
-    sys.stderr.write('\n')
 
-for build in builds:
-    # Version in COPR contains VERSION-RELEASE string. We need just the release.
-    _release = build['source_package']['version'].split('-')[-1]
-    if _release.startswith(ENV_VARS['PKG_RELEASE']):
-        print(build['id'])
-        break
-else:
-    sys.stderr.write('Error: The build with the required release has not been found: {}\n'
-                     .format(ENV_VARS['PKG_RELEASE']))
-    exit_fail()
+def get_builds(ownername, projectname, configpath, client=None, debug=False):
+    client = client or copr.v3.Client(copr.v3.config_from_file(path=configpath))
+    builds = client.build_proxy.get_list(status='succeeded',
+                                         pagination={'order': 'id', 'order_type': 'DESC'},
+                                         ownername=ownername,
+                                         projectname=projectname,
+                                         packagename=ENV_VARS['COPR_PACKAGE'])
+    if debug:
+        json.dump(builds, sys.stderr, sort_keys=True, indent=2)
+        sys.stderr.write('\n')
+    return builds
+
+
+def get_latest_build(ownername, projectname, configpath, match_criteria, client=None, debug=False):
+    client = client or copr.v3.Client(copr.v3.config_from_file(path=configpath))
+    builds = get_builds(ownername, projectname, configpath, client, debug)
+    for build in builds:
+        # Version in COPR contains VERSION-RELEASE string. We need just the release.
+        full_name = '{}-{}'.format(build['source_package']['name'], build['source_package']['version'])
+        release = build['source_package']['version'].split('-')[-1]
+        if re.match(match_criteria, full_name) or release.startswith(match_criteria):
+            return build['id']
+    return None
+
+
+if __name__ == "__main__":
+    # Vet arguments in ENV first
+    ownername = ENV_VARS['COPR_OWNER']
+    projectname = ENV_VARS['COPR_PROJECT']
+    if ENV_VARS['COPR_REPO']:
+        if '/' in ENV_VARS['COPR_REPO']:
+            ownername, projectname = ENV_VARS['COPR_REPO'].split('/', 1)
+        else:
+            projectname = ENV_VARS['COPR_REPO']
+    # If after all those actions either owner or project is not defined - give up and fail
+    if not ownername or not projectname:
+        error = ('Error: Use either COPR_REPO env var in a format "owner/project" or '
+                 'COPR_OWNER & COPR_PROJECT env vars to specify the Copr repository to search in.')
+        _fail(error)
+    # Check that match criteria is defined
+    match_criteria = ENV_VARS['REGEX'] or ENV_VARS['PKG_RELEASE']
```

Comment:
@fernflower maybe the script should print at least a warning or somethine, in case the REGEX + PKB_RELEASE are set in the same time, as in such a case only the REGEX is used. wdyt?

Reply:
Agreed, should be like this now:
```
PKG_RELEASE='12321' REGEX='leapp-repository.*master.*' COPR_REPO='@oamg/leapp' python utils/get_latest_copr_build
Warning: Both REGEX and PKG_RELEASE were set - REGEX has higher priority
3289015
```